src/displayBackend/wayland/Registry.hpp

```diff
@@ -48,11 +49,24 @@ class Registry
 	/**
 	 * Binds registry
 	 */
-	void* bind(const wl_interface *interface)
+	template<typename T>
+	T bind(const wl_interface *interface)
 	{
-		return wl_registry_bind(mWlRegistry, mId, interface, mVersion);
+		checkPointer<T>();
+		T ret = nullptr;
+		if(interface){
+			ret = static_cast<T>(wl_registry_bind(mWlRegistry, mId, interface, mVersion));
+		}
+		return ret;
 	}
 
+   	/**
+	 * The template to identify if 'T' is pointer or not. If not a pointer - compilation error.
```

Comment:
It is pity, that we can't use concepts yet. But at least there is `std::enable_if` and `std::is_pointer` so there is no need for this magic.